backend/src/main/java/com/example/backend/controller/GoalsController.java

```diff
@@ -43,4 +43,29 @@ public ResponseEntity<?> createGoalElement(@RequestParam String goalName, @Reque
         GoalElement savedElement = goalElementRepository.save(element);
         return ResponseEntity.created(new URI("/goals/elements/" + savedElement.getId())).body(savedElement);
     }
+
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity deleteGoal(@PathVariable Long id){
+        User user = userRepository.findTopByOrderByIdAsc();
+        Goal goal = goalRepository.getById(id);
+        goalRepository.deleteById(id);
+        List<GoalElement> to_delete = goalElementRepository.findGoalElementsByGoal(goal);
+        for(GoalElement subgoal: to_delete){
+            goalElementRepository.deleteById(subgoal.getId());
+        }
+        return ResponseEntity.ok().build();
+    }
+
+
+    @PutMapping("/{id}/{goalName}")
+    public ResponseEntity updateGoal(@PathVariable Long id, @PathVariable String goalName){
+        User user = userRepository.findTopByOrderByIdAsc();
+        Goal goal = goalRepository.getById(id);
+        goal.setGoalName(goalName);
+        goal.setUser(user);
+        goalRepository.save(goal);
+        return ResponseEntity.ok(goal.getGoalName());
+    }
+
```

Comment:
Generalnie działa, więc jest git. Jeśli chcemy zachować taki sam sposób wysyłania zapytań do bazy z frontu, to można goalName z @PathVariable zamienić na @RequestParam, bo tak samo jest w edycji wydatków

Reply:
Okej, zaraz wprowadzę zmiany, będzie jednoliciej